src/client/datascience/editor-integration/codewatcher.ts

```diff
@@ -1003,6 +1032,15 @@ export class CodeWatcher implements ICodeWatcher {
             await this.dataScienceErrorHandler.handleError(err);
         }
 
+        if (!result) {
+            // If our cell result was a failure (but not an exception) show an error
+            // for the count of cells cancelled
+            await this.addErrorMessage(file, leftCount);
+
+            // Throw to break out of the promise chain
+            throw new InteractiveCellResultError();
```

Comment:
Slight difference from something like submitCodeImpl as we want to abort the entire chain not just for exceptions (which we want to bubble up) but also for a error in the code, which returns as a false value from .addCode.